### PR TITLE
Player HitBox Shrunk - Includes change to texture loading

### LINKED ISSUE #30: Junkership hitbox should be slightly smaller than the visible sprite

### PR DESCRIPTION
**Pull request by Phyoog**
Closes #30; So! My attempts to create a smaller Junkership hitBox required either basing its size dynamically off the existing rectangle for the Junkership Sprite or hardcoding its size. Hardcoding things is generally a bad idea, but I found that the attributes the hitBox had to be based on weren't loading at the right time. Looking it up showed that we were using the "convenience" method of texture loading and we were sort of locked up as a result.

So I rewrote the texture loading to the preferred method, which fixed MOST of the problem. I've confirmed via testing in the console that the problem now fixes itself if we wait for a few frames to create the first Junkership. I could have hard-coded in a delay, but the best fix is going to be to have the Junkership spawn when its controls are activated, which we have to do anyway. That's getting a bit far afield, so I'd like to merge what I've got here first.

**As for the actual changes...** I made the default Junkership red, since that's the Player 1 color. This is good, because it revealed that the different shapes of the Junkerships makes them feel better with different hitBoxes. That seemed unbalancing though, so I went for a middle ground "best fit" option. I found that speeding up the bad guys also helped make the collision feel more "natural," but they may be a bit fast, considering it'll only speed up with rage mode.

Playable build at: http://mocsarcade.github.io/starjunk/hitBox/

Build using the blue ship to illustrate what I mean by the hitboxes feeling different: http://mocsarcade.github.io/starjunk/hitBox-blueShip/

### CODE REVIEW

File: source/Junkership.js [hunk 1]
```diff
@@ -5,18 +5,24 @@ import Reference from "./Reference.js"
...
 export default class Junkership extends Pixi.Sprite {
     constructor() {
-        super(junkerTex)
+        super(PIXI.loader.resources.redJunkership.texture)
         game.playerCount++
         this.speed = 60
         this.score = new Score()
+
+        this.hitBox = new Pixi.Rectangle( this.x + 1 , this.y + 1 ,
+            this.width - 3 , this.height - 3 )
```
Comment (ehgoodenough) [inline source/Junkership.js]: So at this point, why aren't we using `this.anchor` yet? This would make this math much less confusing.

File: source/Textures.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        .add("redJunkership",    require("./images/red-starship.png"))
+        .add("yellowJunkership", require("./images/yellow-starship.png"))
+        .add("greenJunkership",  require("./images/green-starship.png"))
+        .add("blueJunkership",   require("./images/blue-starship.png"))
+        .add("projectile",       require("./images/projectile.png"))
+        .add("trashbot",         require("./images/enemy-starship.png"))
+        .load()
+
+        .on("complete",setup)
+}
```
Comment (ehgoodenough) [inline source/Textures.js:14]: This is neat! What advantages does this have over the old way of loading textures?

File: source/Textures.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+function setup() {
+    game.spawnWave()
```
Comment (ehgoodenough) [inline source/Textures.js:17]: We could also put generate Junkership here (or allow players to activate Junkerships here) so that there is never a case the Junkership is instantiated before their texture is loaded (and thus avoid the `if (this.width === 1)` kludge.)

File: source/Trashbot.js [hunk 4]
```diff
@@ -27,7 +25,7 @@ export default class Trashbot extends Pixi.Sprite {
                     child.onCollision(this)
                 }
             } else if (child instanceof Junkership) {
-                if (Utility.hasCollision(this, child)) {
+                if (Utility.hasCollision(this, child.hitBox)) {
```
Comment (ehgoodenough) [inline source/Trashbot.js:28]: See what good loose coupling we had here? We didn't have to change `Utility.hasCollision()` at all, only the parameters we passed it. :)

Comment (ehgoodenough) [inline source/Trashbot.js:28]: (For the most part)

File: source/index.js [hunk 5]
```diff
@@ -16,11 +15,8 @@ renderer.roundPixels = true
 var mount = document.getElementById("mount")
 mount.insertBefore(renderer.view, mount.firstChild)
 window.game = new GameContainer()
-Score.resetAll()
-game.addChild(new Junkership())
-
-game.spawnWave()
 
+Textures.initTex()
```
Comment (ehgoodenough) [inline source/index.js]: Could we put this is GameContainer? Maybe?

File: source/Junkership.js (hunk 1 above)
Comment (Phyoog) [inline source/Junkership.js]: We can, but it won't clean up the math. hitBox is a Rectangle, not a Sprite, so it -has- to be drawn from the top left. Changing the origin of the Junkership will make the math even uglier, IMO. Unless I'm misunderstanding your meaning or the mechanics.

I can, however, pull those out to declared constants to minimize the magic in the numbers, if you think that would help.

File: source/Textures.js (hunk 2 above)
Comment (Phyoog) [inline source/Textures.js:14]: Centralizing is always good, IMO. Easier to update filenames and structure if we need to, for one.

Another is the ability to run middleware before or after each module (mostly useful for spritesheets, but I've also been led to believe that we can use that to apply a "rage mode red" tint to the Trashbots, for instance.) Also gives us some built in error handling, progress checks, etc. I believe the memory usage is a little better too, but that's not straight from the horse's mouth, so grain of salt.

Basically, just a more robust utility in general.

File: source/Textures.js (hunk 3 above)
Comment (Phyoog) [inline source/Textures.js:17]: Tried that. Spawning the Junkership here caused a slightly different problem. We had one correctly spawning Junkership (because it waited until it had a texture), but the game checked playerCount (don't ask me how) before the correctly spawned Junkership existed, so it spawned an incorrectly loaded (invisible!) one anyway (via the gameOver() mechanic)

File: source/index.js (hunk 5 above)
Comment (Phyoog) [inline source/index.js]: Shouldn't be a problem. I'll move it in the next commit.

**Comment by Phyoog** [comment]
Playable builds updated. Same links.

**Comment by ehgoodenough** [comment]
All looks good to me! Let's get this merged!
